src/util/handlers/WorkerHandler.ts

```diff
@@ -0,0 +1,44 @@
+import type { ClusterManager } from '../../init/ClusterManager';
+import { NotImplementedHttpError } from '../errors/NotImplementedHttpError';
+import { AsyncHandler } from './AsyncHandler';
+
+/**
+ * A composite handler that will only run all supporting handlers sequentially
+ * if they are executed from a workers process and return the value of the
+ * last supported handler.
+ * The `canHandle` check of this handler will always succeed if single threaded,
+ * or if run from a worker process.
+ */
+export class WorkerHandler<TIn, TOut> extends AsyncHandler<TIn, TOut> {
```

Comment:
I only think of this now, but this can just be one class that takes a boolean as input since they are pretty much identical. And replace `isSingleThreadedOrWorker` with `isWorker` and let the class do the 2 checks itself.

Reply:
Yeah, thought about it too, but wasn't sure if you would rather have the Handler Class names speak for themselves or not (for use in the configs)

Reply:
We already have the URI there with the correct name so should be fine.

Reply:
>And replace isSingleThreadedOrWorker with isWorker and let the class do the 2 checks itself.

Since being primary is a `cluster.isMaster` check. Would you prefer:
 * an extra method ClusterMamager.isPrimary() : this keeps the dependency on the `cluster` module inside the ClusterManager
 * an extra dependency on the `cluster` module in Primary/WorkerHandler to check this.

Reply:
> edit: latest push implemented with option 2

Ah darn, I would have suggested option 1 to keep the entire `cluster` dependency nicely contained within a single class. 😄

Reply:
Which is also fine :P Will refactor ;)